src/App.js

```diff
@@ -186,13 +186,30 @@ function App() {
   }, [theme]);
 
   const [checkingAPIs, setCheckingAPIs] = useState(true);
-
+  useEffect(() => {
+    setauth0works(isAuthenticated === true || isAuthenticated === false);
+  }, [isAuthenticated]);
   useEffect(() => {
     const checkAuth = async () => {
       if (isAuthenticated) {
         setIsLoggedIn(true);
-      } else {
+        getTokenSilently()
+          .then(() => {
+            console.log("Auth0 works when the user is logged in");
+          })
+          .catch((e) => console.log(e));
+      } else if (isAuthenticated === false) {
         setIsLoggedIn(false);
+      } else {
+        setTimeout(() => {
+          if (isAuthenticated !== true && isAuthenticated !== false) {
+            setauth0works(false);
+            setInterval(() => {
+              checkAuth();
+              console.log("Calling check auth");
+            }, 5000);
+          }
+        }, 10000);
```

Comment:
timeout is not being cleared anywhere? This can cause serious performance issues!